### PR TITLE
Add opensearch.xml to enable search of std using omnibox

### PR DESCRIPTION
**Pull request by mbarkhau**

### CODE REVIEW

File: lib/std/special/docs/index.html [hunk 1]
```diff
@@ -4,6 +4,8 @@
     <meta charset="utf-8">
     <title>Documentation - Zig</title>
     <link rel="icon" href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAgklEQVR4AWMYWuD7EllJIM4G4g4g5oIJ/odhOJ8wToOxSTXgNxDHoeiBMfA4+wGShjyYOCkG/IGqWQziEzYAoUAeiF9D5U+DxEg14DRU7jWIT5IBIOdCxf+A+CQZAAoopEB7QJwBCBwHiip8UYmRdrAlDpIMgApwQZNnNii5Dq0MBgCxxycBnwEd+wAAAABJRU5ErkJggg==">
+    <link rel="search" href="opensearch.xml" type="application/opensearchdescription+xml" title="std - Zig Search">
```
Comment (daurnimator) [inline lib/std/special/docs/index.html]: Maybe "Zig Library Documentation Search"?

Comment (mbarkhau) [reply lib/std/special/docs/index.html]: Sure

**Comment by mbarkhau** [comment]
Anny way to retrigger the build?

**Comment by andrewrk** [comment]
@mbarkhau don't worry about the CI, this change is harmless.

I'll merge this soon - I just want to hook it up so that it's only in the std lib docs, and not in everybody's auto generated docs.

**Comment by mbarkhau** [comment]
@andrewrk Good point.

**Comment by andrewrk** [comment]
Thanks for the idea! This is live now.

https://github.com/ziglang/www.ziglang.org/commit/3cbeacb804f78f19536bd072f325a865db243a45

I tried the feature out with Firefox, and it doesn't seem to work very well, it's a lot of button presses to activate, and then it puts extra unwanted characters into the search box. :man_shrugging: 

Auto generated docs is still Work-In-Progress. Updating the std lib documentation is still a manual process. #3402

**Pull request closed without merging**